# FIN 4000 — Chapter 6: Risk Aversion and Capital Allocation to Risky Assets
*Risk aversion, utility, and choosing how much to invest in risky assets*

**ANSWER KEY — fully worked, instructor reference**

This notebook is the Python companion to the Chapter 6 Excel template — same data, same formulas, same numbers. Where the Excel workshop has you build a formula in a cell, this notebook has you build it in a line of Python instead.

## Step 1 — Entry Quiz *(3 minutes, individually)*

1. What does a higher risk-aversion coefficient (A) tell you about an investor's indifference curves?
2. Why does the Capital Allocation Line (CAL) have a constant slope, and what does that slope represent?
3. If an investor's optimal allocation to the risky portfolio is greater than 100%, what does that mean they're doing?

---

In [1]:
import numpy as np

## Step 2 — Worked Example *(10 minutes, instructor-led)*

A risky portfolio offers E(r) = 15%, σ = 22%. The risk-free rate is 5%. Find the optimal allocation to the risky asset for an investor with risk-aversion coefficient A = 4.

In [2]:
E_rp, sigma_p, r_f, A = 0.15, 0.22, 0.05, 4

In [3]:
y_star = (E_rp - r_f) / (A * sigma_p**2)
print(f'y* = {y_star:.2%}')

y* = 51.65%


*y* = (0.15−0.05) ÷ (4×0.22²) = 0.10 ÷ 0.1936 = 51.65%. This investor puts about 52% in the risky portfolio and 48% in T-bills.*

## Step 3 — Pair Work *(12 minutes, with a partner)*

**1.** Compute the complete portfolio's expected return and standard deviation using y* = 51.65%: E(rᴄ) = rƒ + y×(E(rₚ)−rƒ), σᴄ = y×σₚ.

In [4]:
E_rc = r_f + y_star * (E_rp - r_f)
sigma_c = y_star * sigma_p
print(f'E(rC)={E_rc:.2%}  sigma_C={sigma_c:.2%}')

E(rC)=10.17%  sigma_C=11.36%


**2.** A less risk-averse investor has A = 2 instead of 4. Recompute y* — what happens to their allocation?

In [5]:
A_low = 2
y_star_low = (E_rp - r_f) / (A_low * sigma_p**2)
print(f'y* (A=2) = {y_star_low:.2%}  <- over 100%: this investor borrows to leverage up')

y* (A=2) = 103.31%  <- over 100%: this investor borrows to leverage up


**3.** A more risk-averse investor has A = 8. Recompute y* once more. What's the pattern as A increases?

In [6]:
A_high = 8
y_star_high = (E_rp - r_f) / (A_high * sigma_p**2)
print(f'y* (A=8) = {y_star_high:.2%}')

y* (A=8) = 25.83%


## Step 4 — Python Pass *(7 minutes)*

- The optimal-allocation formula y* = [E(rₚ)−rƒ] ÷ (A×σₚ²), built from cells so A can be changed freely
- The complete portfolio's E(rᴄ) and σᴄ for whatever y* results
- A quick side-by-side table showing y* for A = 2, 4, 6, 8, 10

In [7]:
A_values = np.array([2, 4, 6, 8, 10])
y_star_values = (E_rp - r_f) / (A_values * sigma_p**2)
for a, y in zip(A_values, y_star_values):
    print(f'A={a:>2}:  y* = {y:.1%}')

A= 2:  y* = 103.3%
A= 4:  y* = 51.7%
A= 6:  y* = 34.4%
A= 8:  y* = 25.8%
A=10:  y* = 20.7%


## Step 5 — Discussion *(3-5 minutes, no calculation)*

> As an investor's risk-aversion coefficient A rises, their optimal allocation to the risky asset y* falls. Does this mean risk-averse investors always end up worse off? Why or why not?